src/components/tables/Store.ts

```diff
@@ -0,0 +1,123 @@
+import { LiveSpecsExtQuery } from 'hooks/useLiveSpecsExt';
+import produce from 'immer';
+import { devtoolsOptions } from 'utils/store-utils';
+import create, { GetState } from 'zustand';
+import { devtools, NamedSet } from 'zustand/middleware';
+
+export interface SelectableTableStore {
+    rows: Map<string, any>;
+    setRows: (val: LiveSpecsExtQuery[]) => void;
+    removeRows: () => void;
+
+    selected: Map<string, any>;
+    setSelected: (
+        val: SelectableTableStore['selected'],
+        isSelected: boolean
+    ) => void;
+    setAllSelected: (isSelected: boolean) => void;
+
+    resetState: () => void;
+}
+
+export const initialCreateStates = {
+    rows: () => {
+        return new Map();
+    },
+    selected: () => {
+        return new Map();
+    },
+};
+
+export const getInitialStateData = (): Pick<
+    SelectableTableStore,
+    'selected' | 'rows'
+> => {
+    return {
+        selected: initialCreateStates.selected(),
+        rows: initialCreateStates.rows(),
+    };
+};
+
+export const getInitialState = (
+    set: NamedSet<SelectableTableStore>,
+    get: GetState<SelectableTableStore>
+): SelectableTableStore => {
+    return {
+        ...getInitialStateData(),
+        setSelected: (val, isSelected) => {
+            set(
+                produce(({ selected }) => {
+                    if (isSelected) {
+                        selected.set(val, null);
+                    } else {
+                        selected.delete(val);
+                    }
+                }),
+                false,
+                'Selected rows changed'
+            );
+        },
+
+        setAllSelected: (isSelected) => {
+            set(
+                produce(({ selected }) => {
+                    if (isSelected) {
+                        const { rows } = get();
+                        rows.forEach((value, key) => {
+                            selected.set(key, null);
+                        });
+                    } else {
+                        selected.clear();
+                    }
+                }),
+                false,
+                'Selected rows changed'
```

Comment:
Since this action string matches that of `setSelected`, it may be difficult to debug via Redux dev tools. The string is valid in both scenarios and there are other methods of debugging that can be leveraged to distinguish the methods, so this is not something that requires action. Merely calling it out for awareness.